caas/kubernetes/provider/k8s.go

```diff
@@ -1338,16 +1338,40 @@ func (k *kubernetesClient) configurePodFiles(
 			if err = pushUniqueVolume(&workloadSpec.Pod.PodSpec, vol, false); err != nil {
 				return errors.Trace(err)
 			}
-			workloadSpec.Pod.Containers[i].VolumeMounts = append(workloadSpec.Pod.Containers[i].VolumeMounts, core.VolumeMount{
-				// TODO(caas): add more config fields support(SubPath, ReadOnly, etc).
-				Name:      vol.Name,
-				MountPath: fileSet.MountPath,
-			})
+			if err := configVolumeMount(
+				container, workloadSpec,
+				core.VolumeMount{
+					// TODO(caas): add more config fields support(SubPath, ReadOnly, etc).
+					Name:      vol.Name,
+					MountPath: fileSet.MountPath,
+				},
+			); err != nil {
+				return errors.Trace(err)
+			}
 		}
 	}
 	return nil
 }
 
+func configVolumeMount(container specs.ContainerSpec, workloadSpec *workloadSpec, volMount core.VolumeMount) error {
+	if container.Init {
+		for i, c := range workloadSpec.Pod.InitContainers {
+			if c.Name == container.Name {
+				workloadSpec.Pod.InitContainers[i].VolumeMounts = append(workloadSpec.Pod.InitContainers[i].VolumeMounts, volMount)
+				return nil
+			}
+		}
+		return errors.Annotatef(errors.NotFoundf("init container %q", container.Name), "configuring volume mount %q", volMount.Name)
+	}
```

Comment:
Instead of a for loop every time, we could make a map of container name->container up in the configurePodFiles() function above and pass that it